Push support for oct 2025 NISAR sample products

sssangha opened the pull request.

sssangha committed:
Push support for oct 2025 NISAR sample products

sssangha committed:
Push support to handle changes in projection between overlapping NISAR frames

sssangha committed:
Remove debugging print statements

ehavazli commented on tools/bin/ariaTSsetup.py:
```diff
@@ -348,7 +349,10 @@ def generate_stack(aria_prod, stack_layer, output_file_name,
     end_range = aria_prod.products[0][0]['slantRangeEnd'][0]
     range_spacing = aria_prod.products[0][0]['slantRangeSpacing'][0]
-    orbit_direction = str.split(os.path.basename(aria_prod.files[0]), '-')[2]
+    if is_nisar_file:
```
this needs to be flipped or changed to `if not is_nisar_file:`

sssangha replied:
Good catch! I fixed the logic

ehavazli commented:
when I run: `ariaTSsetup.py -f "products/*.h5" -l "all" -d DEM/glo_90.dem -m download -nt 8 -w aria_nisar_test`
I am getting the following issues:

This one doesn't kill the process but looks like the data is not correctly mapped to NISAR
```
Traceback (most recent call last):
  File "/Users/havazli/miniforge3/envs/ARIA-tools/bin/export_product.py", line 40, in <module>
    main()
    ~~~~^^
  File "/Users/havazli/miniforge3/envs/ARIA-tools/bin/export_product.py", line 26, in main
    ARIAtools.extractProduct.export_product_worker(*export_product_args)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/havazli/ARIA-tools/tools/ARIAtools/extractProduct.py", line 1195, in export_product_worker
    product_dict = [[j[layers[ilayer]] for j in full_product_dict],
                     ~^^^^^^^^^^^^^^^^
KeyError: 'azimuthAngle'
```
This is probably another issue related to mapping.
```
2025-11-21 12:59:48,191 - ariaTSsetup.py - WARNING - The following 3 date(s) lack solidEarthTide layers: 20251001, 20251025, 20251106
```

This error is the only one that throws and error and kills the process. However, stack files are extracted.
```
Traceback (most recent call last):
  File "/Users/havazli/miniforge3/envs/ARIA-tools/bin/ariaTSsetup.py", line 715, in <module>
    main()
    ~~~~^^
  File "/Users/havazli/miniforge3/envs/ARIA-tools/bin/ariaTSsetup.py", line 704, in main
    generate_stack(
    ~~~~~~~~~~~~~~^
        standardproduct_info, layer, ARIA_STACK_OUTFILES[layer],
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        **stack_dict)
        ^^^^^^^^^^^^^
  File "/Users/havazli/miniforge3/envs/ARIA-tools/bin/ariaTSsetup.py", line 339, in generate_stack
    ARIAtools.util.vrt.get_basic_attrs(dlist[0])
                                       ~~~~~^^^
IndexError: list index out of range
```

ehavazli commented:
I get the error below related to the progress bar creation. I am going to commit a change to overcome this issue. This is not corrected to the aforementioned issues.

```Traceback (most recent call last):
  File "/Users/havazli/miniforge3/envs/ARIA-tools/bin/ariaTSsetup.py", line 715, in <module>
    main()
    ~~~~^^
  File "/Users/havazli/miniforge3/envs/ARIA-tools/bin/ariaTSsetup.py", line 704, in main
    generate_stack(
    ~~~~~~~~~~~~~~^
        standardproduct_info, layer, ARIA_STACK_OUTFILES[layer],
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        **stack_dict)
        ^^^^^^^^^^^^^
  File "/Users/havazli/miniforge3/envs/ARIA-tools/bin/ariaTSsetup.py", line 289, in generate_stack
    prog_bar = ARIAtools.util.misc.ProgressBar(
        maxValue=len(int_list), print_msg='Creating stack: ')
  File "/Users/havazli/ARIA-tools/tools/ARIAtools/util/misc.py", line 41, in __init__
    self.reset()
    ~~~~~~~~~~^^
  File "/Users/havazli/ARIA-tools/tools/ARIAtools/util/misc.py", line 46, in reset
    self.update_amount(0)  # Build progress bar string
    ~~~~~~~~~~~~~~~~~~^^^
  File "/Users/havazli/ARIA-tools/tools/ARIAtools/util/misc.py", line 60, in update_amount
    percentDone = (diffFromMin / float(self.span)) * 100.0
                   ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
ZeroDivisionError: float division by zero
```

ehavazli committed:
ZeroDivisionError bug fix

sssangha committed:
Fix error in orbit_direction assignment for NISAR file(s)

sssangha committed:
Fix bug in handling projection for NISAR geometry layers

sssangha committed:
Implement NISAR azimuthAngle calculation from LOS vectors

Added functionality to compute azimuthAngle from losUnitVectorX and losUnitVectorY. This includes creating temporary raster files, calculating the angle, and updating the output with NoData handling.

sssangha commented:
Added in support for computing `azimuthAngle` on the fly from `losUnitVectorX` and `losUnitVectorY`.

Here is a sample, stitched field for an ascending track product:
<img width="655" height="567" alt="Screenshot 2026-01-08 at 4 25 26 PM" src="https://github.com/user-attachments/assets/dbf31fab-488d-4d56-9358-92c7f9471279" />

sssangha committed:
Adding amplitude support for NISAR product

sssangha committed:
Add wrapped interferogram and modify data keys

Updated data layer dictionary to include wrapped interferogram and additional keys.

sssangha committed:
Refactor projection handling to remove EPSG prefix

sssangha committed:
Update EPSG parameter handling in seq_stitch.py

sssangha committed:
Change epsg type from int to string with EPSG prefix

sssangha committed:
Refactor GDAL Warp calls for better resource management

Refactored GDAL Warp calls to assign output to variables and ensure proper cleanup.

sssangha committed:
Update warning filters in stitch.py

sssangha committed:
Update extract_bperp_dict to extract_bperp_dict_ts

sssangha committed:
Refactor GDAL VRT handling for better resource management

sssangha committed:
Refactor height metadata handling in extractProduct.py

sssangha committed:
Add helper function to get height metadata safely

Added a helper function to safely check heights from metadata.

sssangha committed:
Refactor height metadata handling and file updates

sssangha committed:
Improve dataset management in vrt.py

Refactor VRT file handling and ensure proper dataset closure.

sssangha committed:
Refactor extractProduct.py to handle data opening leaks

Updated comments for clarity and context. Improved resource management with context managers for file handling. Simplified attribute handling and ensured safe closing of resources.

sssangha committed:
Update raster averaging with warnings handling

Refactor raster processing to use warnings and improve clarity.

sssangha committed:
Update ariaTSsetup to support nisar differential layers

Fix code logic to support TROPO and SET layers in NISAR not being distributed as individual dates like in ARIA-S1-GUNW

sssangha committed:
Add condition to check for NISAR files in tropo layer

ehavazli commented:
This PR adds full support for NISAR GUNW products, including extraction, stitching, and stacking of all data and correction layers. The outputs are ready for use with downstream time-series packages.
